root_utils/root_file_utils.py

```diff
@@ -202,19 +204,27 @@ def get_hit_photons(self):
             start_time.append(np.zeros(n_photons, dtype=np.float32))
             start_position.append(np.zeros((n_photons, 3), dtype=np.float32))
             end_position.append(np.zeros((n_photons, 3), dtype=np.float32))
+            start_dir.append(np.zeros((n_photons, 3), dtype=np.float32))
+            end_dir.append(np.zeros((n_photons, 3), dtype=np.float32))
             photons = self.trigger.GetCherenkovHitTimes()
-            end_time[t][:] = [p.GetTruetime() for p in photons]
-            track[t][:] = [p.GetParentID() for p in photons]
-            try:  # Only works with new tracking branch of WCSim
-                start_time[t][:] = [p.GetPhotonStartTime() for p in photons]
-                for i in range(3):
-                    start_position[t][:,i] = [p.GetPhotonStartPos(i)/10 for p in photons]
-                    end_position[t][:,i] = [p.GetPhotonEndPos(i)/10 for p in photons]
-            except AttributeError: # leave as zeros if not using tracking branch
-                pass
+            for it in range(n_photons):
```

Comment:
in the CIDER meeting, we found that looping the `photons` object  multiple time makes a significant CPU time in conversion

Reply:
That's strange, in my testing I found the opposite was true, but that was quite a while ago. It might be related to different python versions optimising differently. Are you using WCTE container version?

If it's better to have a single loop, it should probably be `for p in photons:` or `for it, p in enumerate(photons):`

Reply:
Ran a quick test using ROOT 6.26 and python 3.8.10 and confirmed it does seem faster with the single loop.

For 3000 muons:

Using multiplt list comprehensions:
real    1m39.704s
user    1m38.026s
sys     0m1.094s

Using `for it in range(n_photons)`:
real    1m3.601s
user    1m1.952s
sys     0m1.127s

Using `for it, p in enumerate(photons)`:
real    1m0.813s
user    0m59.190s
sys     0m1.129s

Reply:
We were testing in the cider container, with root_v5.34.38 and probably python 3.6. In the extreme cases of 1500 events of photon simulation and ~120000 true photons per event, it has speeded up by a factor of 3 or 4.